### PR TITLE
WIP: Application

### PR DESCRIPTION
**Pull request by HitoDZ**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -328,3 +328,6 @@ ASALocalRun/
 
 # MFractors (Xamarin productivity tool) working folder 
 .mfractor/
+Mongo/Mongo.Migrations/Defaults.cs
+Api.FacePlusPlus/Defaults.cs
+Api.Flickr/Defaults.cs
```
Comment (vova-lantsov-dev) [inline .gitignore]: Can be changed to single line
```suggestion
Defaults.cs
```

File: Host/Controllers/ImageController.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Threading.Tasks;
+using Domain.Persistance;
+using Host.Repositories;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Host.Controllers
+{
+    public class ImageController : ControllerBase
```
Comment (vova-lantsov-dev) [inline Host/Controllers/ImageController.cs:9]: Add route prefix directly for controller
```suggestion
    [Route(RouteConstants.Base)]
    public class ImageController : ControllerBase
```

File: Host/Repositories/ImageRepository.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        private readonly MongoDbContext _context;
+
+        public ImageRepository(MongoDbContext context)
+        {
+            _context = context;
+        }
+       
+        public async Task<IReadOnlyCollection<DBPhotoModel>> GetAll()
+        {
+            return await _context.Photos.Find(_ => true).ToListAsync();
```
Comment (vova-lantsov-dev) [inline Host/Repositories/ImageRepository.cs]: Optimise code
```suggestion
            return await _context.Photos.Find(FilterDefinition<DBPhotoModel>.Empty).ToListAsync();
```

File: Host/Startup.cs [hunk 4]
```diff
@@ -22,13 +24,13 @@ public Startup(IConfiguration configuration)
 
         public IConfiguration Configuration { get; }
```
Comment (vova-lantsov-dev) [inline Host/Startup.cs]: Remove unused property 'Configuration' and Startup constructor

File: Mongo/Domain/Models/PhotoModel.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace Domain.Models
+{
+    public sealed class PhotoModel
+    {
+        public readonly string Url;
+        public readonly string Name;
+        public readonly IReadOnlyCollection<Emotion> Emotions;
```
Comment (vova-lantsov-dev) [inline Mongo/Domain/Models/PhotoModel.cs]: readonly? What for?

File: Mongo/Mongo.Migrations/Program.cs [hunk 6]
```diff
@@ -49,43 +49,53 @@ internal static async Task Main(string[] args)
...
+                {
+                    Name = photo.Title,
+                    Url = photoUrl,
+                    Emotions = emotions
+                });
+                Console.WriteLine(" '" + photo.Title + " Added'");
+            }
+            
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine("All data ready! :)");
```
Comment (vova-lantsov-dev) [inline Mongo/Mongo.Migrations/Program.cs]:
```suggestion
            Info("All data ready! :)");
```

File: Mongo/Mongo.Migrations/Program.cs [hunk 7]
```diff
@@ -49,43 +49,53 @@ internal static async Task Main(string[] args)
...
                     });
-                }
+                
+                await context.Photos.InsertOneAsync(new DBPhotoModel
+                {
+                    Name = photo.Title,
+                    Url = photoUrl,
+                    Emotions = emotions
+                });
+                Console.WriteLine(" '" + photo.Title + " Added'");
```
Comment (vova-lantsov-dev) [inline Mongo/Mongo.Migrations/Program.cs]:
```suggestion
                Info(" '" + photo.Title + " Added'");
```

File: Mongo/Mongo.Migrations/Program.cs [hunk 8]
```diff
@@ -49,43 +49,53 @@ internal static async Task Main(string[] args)
...
             using (var faceClient = new FacePlusPlusClient(_faceClientOptions))
-                foreach (var photo in flickrResult.Photos)
+            foreach (var photo in flickrResult.Photos)
+            {
+                Console.Write("Processing - " + ++i);
+                if (await context.Photos.Find(p => p.Name == photo.Title).AnyAsync())
                 {
-                    if (await context.Photos.Find(p => p.Name == photo.Title).AnyAsync())
-                        continue;
+                    Console.WriteLine(" '" + photo.Title + " already Exist'");
```
Comment (vova-lantsov-dev) [inline Mongo/Mongo.Migrations/Program.cs]:
```suggestion
                    Info(" '" + photo.Title + " already Exist'");
```

File: Host/Controllers/RouteConstants.cs [hunk 9]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Host.Controllers
+{
+    public static class RouteConstants
+    {
+        private const string Base = "api";
```
Comment (vova-lantsov-dev) [inline Host/Controllers/RouteConstants.cs]:
```suggestion
        public const string Base = "api";
```

File: Host/Controllers/RouteConstants.cs [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+        public const string Images = Base + "/images";
```
Comment (vova-lantsov-dev) [inline Host/Controllers/RouteConstants.cs]:
```suggestion
        public const string Images = "images";
```

File: Host/Startup.cs [hunk 11]
```diff
@@ -22,13 +24,13 @@ public Startup(IConfiguration configuration)
...
 
-        // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+            services.AddMvcCore().AddJsonFormatters().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+            services.AddSingleton<MongoDbContext>(new MongoDbContext("mongodb://localhost"));
```
Comment (vova-lantsov-dev) [inline Host/Startup.cs]:
```suggestion
            services.AddSingleton(new MongoDbContext("mongodb://localhost"));
```

File: Host/Startup.cs [hunk 12]
```diff
@@ -22,13 +24,13 @@ public Startup(IConfiguration configuration)
...
+            services.AddTransient<ImageRepository>();
```
Comment (vova-lantsov-dev) [inline Host/Startup.cs:24]: We can use MongoDbContext without this repository. Or you would like to use repository?

**Review by vova-lantsov-dev — CHANGES_REQUESTED** [review]

File: Host/Startup.cs (hunk 12 above)
Comment (HitoDZ) [reply Host/Startup.cs:24]: For now I will not change it due to netural changes.

**Review by vova-lantsov-dev — APPROVED** [review]

**Pull request merged**